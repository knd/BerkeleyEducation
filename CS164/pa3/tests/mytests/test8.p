% fill the country map so that no 2 regions have same color

adj(1,2).  adj(2,1). 
adj(1,3).  adj(3,1). 
adj(1,4).  adj(4,1). 
adj(1,5).  adj(5,1). 
adj(2,3).  adj(3,2). 
adj(2,4).  adj(4,2). 
adj(3,4).  adj(4,3). 
adj(4,5).  adj(5,4). 
fill(1,red,a).     fill(1,red,b). 
fill(2,blue,a).    fill(2,blue,b). 
fill(3,green,a).   fill(3,green,b). 
fill(4,yellow,a).  fill(4,blue,b). 
fill(5,blue,a).    fill(5,green,b).
oops(C) :- adj(X,Y), fill(X,Color,C), fill(Y,Color,C). 
oops(A,B,C) :- adj(A,B), fill(A,Color,C), fill(B,Color,C). 
oops(A,B,b)?
