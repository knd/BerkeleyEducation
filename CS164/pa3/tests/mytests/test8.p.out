Asking for solution 1
A = 2
B = 4
Asking for solution 2
A = 4
B = 2
Asking for solution 3
None
